### DISCUSSION TITLE
NEETBOX frontend on its way

### DISCUSSION THREAD

**Discussion by visualDust** [visualDust/neetbox#77] [Announcements]
Following features are availiavble in the next "stable" release

| feature | python api | server | frontend |
| :-----: | :-----: | :-----: | :-----: |
| logging | :heavy_check_mark: ` from neetbox import logger`  |  :heavy_check_mark: api/db support | :heavy_check_mark: show colorful logs from client |
| hardware status | :heavy_check_mark: optional in config file neetbox.toml |  :heavy_check_mark:  api/db support |  :heavy_check_mark:  show cpu/gpu/ram usage line chart |
| platform info | :heavy_check_mark: optional in config file neetbox.toml |  :heavy_check_mark:  api/db support |  :heavy_check_mark:  show system informations of running machines with neetbox |
| remote code run | :heavy_check_mark: `from neetbox import action` | :heavy_check_mark: api | :heavy_check_mark: buttons to run actions, and pop ups to fillup arguments, also show result from client |
| show images | :heavy_check_mark: `from neetbox import add_image` | :heavy_check_mark: api/db support | :heavy_check_mark: show images |
| show image from tensor | :heavy_check_mark: `from neetbox import add_image` | :heavy_check_mark: api/db support | :heavy_check_mark: show images from tensor |
| show image from matplotlib | :heavy_check_mark: `from neetbox import add_figure` | :heavy_check_mark: api/db support | :heavy_check_mark: show images from matplotlib |
| show custom line chart |  :heavy_check_mark: `from neetbox import add_scaler`  | :heavy_check_mark: api/db support | :heavy_check_mark: show linechart from client |
| history | - | :heavy_check_mark: stores everything from client using sqlite, per db per client | :heavy_check_mark: fetch history from server |
| gather multiple client in one frontend | - | :heavy_check_mark: multi client and multi frontend connections are allowed | :heavy_check_mark: show pages for multiple projects running on multiple machine |
| distinguish different runs | :heavy_check_mark:  auto new run id on each start | :heavy_check_mark:  added indexed db column run id | :heavy_check_mark: users can select runs on page |

neetbox cli:
```
Usage: neet [OPTIONS] COMMAND [ARGS]...

Options:
  -v, --verbose BOOLEAN  set verbose flag to True
  --help                 Show this message and exit.

Commands:
  init      initialize current folder as workspace (generate config file)
  list      Show list of connected project names
  serve     serve neetbox server in attached mode
  shutdown  shutdown neetbox server on specific port
  status    Show running project status of given name
```

frontend supports darkmode and lightmode. screenshots:
![image](https://github.com/visualDust/neetbox/assets/33346934/516352f8-cfeb-4c90-ac1a-e61fa58576d5)
![image](https://github.com/visualDust/neetbox/assets/33346934/1bd54f03-05cf-4372-a527-3f3a695d0e66)

**Comment by therainisme** [comment]
Wow! It's too amazing! When is the stable release?

**Reply by visualDust** [reply]
not sure, depends on frontend dev. ask @lideming for frontend things

**Comment by PommesPeter** [comment]
wow, amazing
